nixos/tests/mysql/mariadb-galera.nix

```diff
@@ -0,0 +1,290 @@
+import ./../make-test-python.nix ({ pkgs, ...} :
+
+let
+  mysqlenv-common      = pkgs.buildEnv { name = "mysql-path-env-common";      pathsToLink = [ "/bin" ]; paths = with pkgs; [ bash gawk gnutar inetutils which ]; };
+  mysqlenv-mariabackup = pkgs.buildEnv { name = "mysql-path-env-mariabackup"; pathsToLink = [ "/bin" ]; paths = with pkgs; [ gzip iproute netcat procps pv socat ]; };
+  mysqlenv-rsync       = pkgs.buildEnv { name = "mysql-path-env-rsync";       pathsToLink = [ "/bin" ]; paths = with pkgs; [ lsof procps rsync stunnel ]; };
+
+in {
+  name = "mariadb-galera";
+  meta = with pkgs.stdenv.lib.maintainers; {
+    maintainers = [ izorkin ];
+  };
+
+  nodes = {
+    galera_01_maria =
+      { pkgs, ... }:
+      {
+      networking = {
+        interfaces.eth1 = {
+          ipv4.addresses = [
+            { address = "192.168.1.1"; prefixLength = 24; }
+          ];
+        };
+        firewall.enable = true;
+        firewall.allowedTCPPorts = [ 3306 4444 4567 4568 ];
+        firewall.allowedUDPPorts = [ 4567 ];
+      };
+      users.users.testuser = { };
+      systemd.services.mysql = with pkgs; {
+        path = [ mysqlenv-common mysqlenv-mariabackup ];
+      };
+      services.mysql = {
+        enable = true;
+        package = pkgs.mariadb;
+        ensureDatabases = [ "testdb" ];
+        ensureUsers = [{
+          name = "testuser";
+          ensurePermissions = {
+            "testdb.*" = "ALL PRIVILEGES";
+          };
+        }];
+        initialScript = pkgs.writeText "mariadb-init.sql" ''
+          GRANT ALL PRIVILEGES ON *.* TO 'check_repl'@'localhost' IDENTIFIED BY 'check_pass' WITH GRANT OPTION;
+          GRANT ALL PRIVILEGES ON *.* TO 'check_repl'@'192.168.1.1' IDENTIFIED BY 'check_pass' WITH GRANT OPTION;
+          GRANT ALL PRIVILEGES ON *.* TO 'check_repl'@'192.168.1.2' IDENTIFIED BY 'check_pass' WITH GRANT OPTION;
```

Comment:
Other tests seem to refer to these IPs via things like
```
(lib.head nodes.galera_02_maria.config.networking.interfaces.eth1.ipv4.addresses)
```

I'm not sure if it gets too messy - maybe we can add these lines to `/etc/hosts` and use hostnames here?
> (lib.head config.networking.interfaces.eth1.ipv4.addresses).address